src/GShark/Geometry/Mesh.cs

```diff
@@ -548,6 +548,127 @@ private FaceData CountFaceEdges()
+                Vector3 t = Vector3.CrossProduct(edges_vectors[i], w);
+                double dotProduct = n * t;
+                dotProducts.Add(dotProduct);
+            }
+
+            return ((dotProducts[0] > 0) && (dotProducts[1] > 0) && (dotProducts[2] > 0));
+            }
+         
+        // Function that finds the closest point to a triangle (given the 3 vertices as points)
+        private Point3 ClosestPointToTriangle(Point3[] trianglePoints, Point3 point)
+        {
+            if (trianglePoints.Length != 3)
+                throw new ArgumentOutOfRangeException($"{nameof(trianglePoints)} must be 3.");
+            // Creates Plane from 3 points
+            // TODO: checks if the triangle degenerate into a line or a point before this
+            Plane facePlane = new Plane(trianglePoints[0], trianglePoints[1], trianglePoints[2]);
+            // Find projection of point of the plane
+            Point3 closestPoint = facePlane.ClosestPoint(point, out _);
+            // if the projection of the point is inside the triangle, we return the closest point
+            if (IsPointInTriangle(closestPoint, trianglePoints))
+                return closestPoint;
+
+            // If not, we create a line for each triangle edges
+            // and then find the closest point on each edge
+            Line segment_AB = new Line(trianglePoints[0], trianglePoints[1]);
+            Line segment_BC = new Line(trianglePoints[1], trianglePoints[2]);
+            Line segment_CA = new Line(trianglePoints[2], trianglePoints[0]);
+            List<Line> segments = new List<Line>() { segment_AB, segment_BC, segment_CA };
+            List<Point3> edgesClosestPoints = new List<Point3>();
+            foreach (Line segment in segments)
+            {
+                Point3 segmentClosestPoint = segment.ClosestPoint(point);
+                edgesClosestPoints.Add(segmentClosestPoint);
+            }
+            return point.CloudClosestPoint(edgesClosestPoints);
+        }
+
+        public Point3 ClosestPoint(Point3 point)
+        {
+            List<MeshVertex> meshVertices = Vertices;
+
+            //All faces check 
+            System.Collections.Generic.IEnumerable<MeshFace> AllFaces = this.Faces;
+
+            var CloserPointToFace_List = new List<Point3>();
+
+            foreach (MeshFace face in AllFaces)
+            {
+                // Retrives the Vertices associated with each face
+                List<MeshVertex> GSVertices = face.AdjacentVertices();
+                // Converts from MeshVertex to Point3
+                List<Point3> FacePoints = GSVertices.ConvertAll(v => (Point3)v);
+
+                if (GSVertices.Count == 3)
+                {
+                    var trianglePoints = new Point3[] { FacePoints[0], FacePoints[1], FacePoints[2] };
+                    Point3 ClosestPoint0 = ClosestPointToTriangle(trianglePoints, point);
+                    CloserPointToFace_List.Add(ClosestPoint0);
```

Comment:
ALL variables should always be `camelCase`.
~~`CloudPoint0`~~ -> `cloudPoint0`
~~`CloserPointToFace_List`~~ -> `closerPointToFaceList`

Do this for all variables of entire code!